cypress/integration/channel/message_edit_post_with_attachment_spec.js

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 3] */
+
+describe('MM-13697 Edit Post with attachment', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+    });
+
+    it('Pasted text should be pasted where the cursor is', () => {
+        // 2. Got to a test channel on the side bar
+        cy.get('#sidebarItem_town-square').should('be.visible').click({force: true});
+
+        // * Validate if the channel has been opened
+        cy.url().should('include', '/ad-1/channels/town-square');
+
+        // 3. Attach image
+        cy.uploadFile('#fileUploadButton input', '../fixtures/mattermost-icon.png', 'image/png');
+
+        // 4. Type 'This is sample text'
+        cy.get('#post_textbox').type('This is sample text').should('contain', 'This is sample text');
+
+        // 5. Submit post
+        cy.get('#create_post').submit();
+
+        // 6. Get last post ID
+        cy.getLastPostId().then((postID) => {
+            // 7. click  dot menu button
+            cy.clickPostDotMenu();
+
+            // 8. click edit post
+            cy.get(`#edit_post_${postID}`).click();
+
```

Comment:
Inserting text of "add " after 4 left arrows is not giving me consistent outcome.  Sometimes, only 2 left arrows took effect.  I'd suggest to wrap it with:
```javascript
cy.focused().then(($el) => {
    // * Check if focus is set to "edit_textbox"
    assert.equal($el[0].id, 'edit_textbox');

    // 9. Edit message to 'This is sample add text'
    ...
    cy.get(`#${postID}_message`).find('.file-view--single').should('be.visible');
});
```

That way, we're also checking if focus is correctly set to "edit_textbox".

Reply:
can you explain to me, what's different between wrapping with focus and without wrapping.

Reply:
Without wrapping, typing to '#edit_textbox' is executed immediately even if the focus is not yet on '#edit_textbox'.  As a result, sometimes not all in `.type('{leftarrow}{leftarrow}{leftarrow}{leftarrow}add ')` is placed in the '#edit_textbox'.

Wrapping with `cy.focused()` let us type not only on present but focused  element, plus a bonus of checking where the focus is.

Reply:
thanks @saturninoabril for your clear explanation